### PR TITLE
Android Support - Introduce launch argument for Mock Server (EXPOSUREAPP-12689)

### PR DESCRIPTION
**Pull request by jurajkusnier**
### Testing
It should be possible to change environment using command line parameters
example:
```bash
adb shell am start -n de.rki.coronawarnapp.test/de.rki.coronawarnapp.ui.launcher.LauncherActivity -e env wru
adb shell am start -n de.rki.coronawarnapp.test/de.rki.coronawarnapp.ui.launcher.LauncherActivity -e env_base64 ewogICAgIlVTRV9FVVJfS0VZX1BLR1MiOiB0cnVlLAogICAgIlNVQk1JU1NJT05fQ0ROX1VSTCI6ICJodHRwczovL3N1Ym1pc3Npb24tY2ZmNGY3MTQ3MjYwLmNvcm9uYXdhcm4uYXBwIiwKICAgICJET1dOTE9BRF9DRE5fVVJMIjogImh0dHBzOi8vc3ZjOTB3cnUubWFpbi5weC50LW9ubGluZS5kZSIsCiAgICAiVkVSSUZJQ0FUSU9OX0NETl9VUkwiOiAiaHR0cHM6Ly92ZXJpZmljYXRpb24tMjU1NGU2OGIuY29yb25hd2Fybi5hcHAiLAogICAgIkRBVEFfRE9OQVRJT05fQ0ROX1VSTCI6ICJodHRwczovL2RhdGEtaDZreGY5b2UuY29yb25hd2Fybi5hcHAiLAogICAgIkxPR19VUExPQURfU0VSVkVSX1VSTCI6ICJodHRwczovL2xvZ3VwbG9hZC1oNmt4ZjlvZS5jb3JvbmF3YXJuLmFwcCIsCiAgICAiRENDX1JFSVNTVUFOQ0VfU0VSVkVSX1VSTCI6ICJodHRwczovL2FwaS5yZWNlcnRpZnkuZGVtby51YmlyY2guY29tIiwKICAgICJTQUZFVFlORVRfQVBJX0tFWSI6ICJBSXphU3lCTFBROHBiQkxUUThldkpmaDR1M0FBdmlkSEVEU2xlaDAiLAogICAgIlBVQl9LRVlTX1NJR05BVFVSRV9WRVJJRklDQVRJT04iOiAiTUZrd0V3WUhLb1pJemowQ0FRWUlLb1pJemowREFRY0RRZ0FFM0JZVHhyMkh1SllRRytkN0V6dTZLUzhHRWJGa2lFdnlKRmcwaitDODM5Z1RqVDZqN0hvMEVYWFovYTA3WmZ2S2NDMmNtYzFTdW5zcnFVOUpvdjFKNVE9PSIsCiAgICAiQ1JPV0RfTk9USUZJRVJfUFVCTElDX0tFWSI6ICJQSEpQdHRpT0pzQjNIMG5zemFWQTNZWVBTU0VhSnBzZ2NKSmd1a2NvWE45UTVlbTVGbkdMYmFDMFhpQ3MzV1FXTGtNb1AzTmdFWHRjZ29WQUdTb1dBWmY0azQtLXNFMktIa3FiNEVWWWlyMWhJODl2c1E4YXlGMlBrZGpoOUJrRCIsCiAgICAiRENDX1NFUlZFUl9VUkwiOiAiaHR0cHM6Ly9kY2MtY2ZmNGY3MTQ3MjYwLmNvcm9uYXdhcm4uYXBwIgogIH0=
```

### Jira Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12689

### CODE REVIEW

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragmentViewModel.kt [hunk 1]
```diff
@@ -23,7 +23,18 @@ class DebugOptionsFragmentViewModel @AssistedInject constructor(
     val environmentState = environmentStateFlow.asLiveData(context = dispatcherProvider.Default)
     val environmentStateChange = SingleLiveEvent<EnvironmentState>()
 
-    fun selectEnvironmentTytpe(type: String) {
+    fun clearLaunchEnvironment() {
+        envSetup.launchEnvironment = null
+        envSetup.toEnvironmentState()
+            .copy(isOverwritten = false)
+            .let {
+                environmentStateFlow.value = it
```
Comment (schauersbergern) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragmentViewModel.kt:31]: Maybe also refresh view (hide clear button, show toggleGroup) when launch environment is cleared

Comment (jurajkusnier) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/debugoptions/ui/DebugOptionsFragmentViewModel.kt:30]: test menu is fixed

**Review by schauersbergern — DISMISSED** [review]
Code looks good, works on my phone (Pixel 6 Android 12)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/launcher/LauncherActivityViewModel.kt [hunk 2]
```diff
@@ -104,6 +113,36 @@ class LauncherActivityViewModel @AssistedInject constructor(
...
+    }
+
+    fun setEnvironment(environment: LauncherParameter.Base64Environment) {
+        try {
+            environment.value.decodeBase64()?.toByteArray()?.let {
+                envSetup.launchEnvironment = objectMapper.readTree(it)
+            }
+            clearAndRestart()
+        } catch (e: Exception) {
+            Timber.e(e, "Can't set environment data using $environment")
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/launcher/LauncherActivityViewModel.kt:134]: Same here "Invalid env and add e.message ?"?

**Review by mtwalli — COMMENTED** [review]
atm when changing the env with a faulty one , the toast message is confusing as it fallback to the previous one , maybe specific error message with more info would be helpful ?

**Review by mtwalli — DISMISSED** [review]
LGTM, Thanks

**Review by schauersbergern — DISMISSED** [review]

**Review by schauersbergern — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**